O/ONNXRuntime/build_tarballs.jl

```diff
@@ -122,11 +135,12 @@ dependencies = [
         compat = tensorrt_compat,
         platforms = cuda_platforms),
     Dependency("Zlib_jll"; platforms = cuda_platforms),
-    HostBuildDependency(PackageSpec("protoc_jll", Base.UUID("c7845625-083e-5bbe-8504-b32d602b7110"), v"3.16.1"))
+    BuildDependency(PackageSpec("CUDA_full_jll", first(cuda_versions)); platforms = cuda_platforms),
```

Comment:
CUDA_full is deprecated and should be replaced by CUDA_SDK (and CUDA_SDK_static if needed).

Reply:
I would like to (at first, in this PR - and in #4386) build this old version of ONNXRuntime, v1.10.0, to get support for older Nvidia Jetson devices ("JetPack 4"), e.g. Jetson Nano - which requires using CUDA 10.2 (and CUDNN v8.2 and [TensorRT v8.0](https://github.com/JuliaBinaryWrappers/TensorRT_jll.jl/releases/tag/TensorRT-v8.0.1%2B0)). Also, TensorRT v8.0 was only available for CUDA 11.3, and not 11.4, so for CUDA 11 I was forced to use CUDA 11.3 and not 11.4. It seems I then have to use CUDA_full and not CUDA_SDK, right?

What I am aiming for is getting as close as possible to the build versions listed in the tables on https://onnxruntime.ai/docs/execution-providers/CUDA-ExecutionProvider.html#requirements and https://onnxruntime.ai/docs/execution-providers/TensorRT-ExecutionProvider.html#requirements - while being able to use the binaries provided for the Jetson (aarch64) devices listed on https://elinux.org/Jetson_Zoo#ONNX_Runtime (i.e. v1.10 and v1.11) (and the binaries for Windows from e.g. https://github.com/microsoft/onnxruntime/releases/tag/v1.10.0)

Reply:
Yeah CUDA_SDK only supports 11.4 and upwards.

Reply:
How to best support this CUDA 10.2 scenario? I know that CUDA 10.2 support was dropped for CUDA_Runtime >= v0.3 (and CUDA.jl >= v4.1)...

I have also noticed that CUDA platform tag changed from including minor versions to specifying e.g. `cuda=11.0` or `cuda=12.0` for later versions...?

Reply:
... so for ONNXRuntime v1.10-v1.11 (the versions supporting JetPack 4.6, i.e. CUDA 10.2), we have to stick with CUDA_full, right?

And then from ONNXRuntime v1.12, replace CUDA_full with with CUDA_SDK (and perhaps CUDA_SDK_static as a BuildDependency, to have the static cudart for CMake/CUDA configuration).